### PR TITLE
Update honey badger

### PR DESCRIPTION
**Pull request by Ch4s3**
Updated to latest Honey Badger based on http://docs.honeybadger.io/article/197-upgrading-to-honeybadger-2-0.

### CODE REVIEW

**Comment by dwradcliffe** [comment]
Why did you rename the initializer?

Also, it looks like we can skip the yml and just use ENV for the api key?

**Comment by arthurnn** [comment]
unfortunately we are locked on version 1. as we cannot do this on version 2: https://github.com/rubygems/rubygems.org/blob/master/config/initializers/honeybadger.rb#L4-L6

**Comment by Ch4s3** [comment]
@dwradcliffe I followed the honey badger instructions. 
@arthurnn do you know if this functionality is completely gone in 2.x, or is it possibly just different? I haven't used `config.ignore` in HoneyBadger.

**Comment by Ch4s3** [comment]
@arthurnn It looks like [ignore](http://docs.honeybadger.io/article/198-gem-configuration) can be set up in the yaml. Or does it need to happen live and that's the issue?

**Comment by arthurnn** [comment]
cc @joshuap as he will know better.
But this is related to https://github.com/honeybadger-io/honeybadger-ruby/issues/116

**Comment by Ch4s3** [comment]
Ok, I read through all of that and the update to honeybadger proper. If the intent of [this](https://github.com/rubygems/rubygems.org/blob/master/config/initializers/honeybadger.rb#L4-L6) is to "Ignore all errors on Honeybadger that rails handles", the that list could be set in the Yaml.

**Comment by Ch4s3** [comment]
I'm not sure if I got this right, but according to their 2.x [docs](http://docs.honeybadger.io/article/42-ignoring-exceptions-by-class) you can do this with custom logic. I tried to do that [here](https://github.com/Ch4s3/rubygems.org/blob/26a616c8e991ac7d9c443bedb300436f9a5beaff/config/initializers/honeybadger.rb), but I'm unsure if it is correct.

**Comment by joshuap** [comment]
@arthurnn @Ch4s3 for what it's worth 2.0 has a current list of rescue responses from Rails (as of 2.0.5), so it's not strictly necessary to ignore these programatically. If you're concerned about future updates to Rails, then you could indeed still use the `Honeybadger.exception_filter` callback to do this, but I think you need to tweak 26a616c8e991ac7d9c443bedb300436f9a5beaff a bit. This should work:

```ruby
Honeybadger.exception_filter do |notice|
  !ActionDispatch::ExceptionWrapper.rescue_responses.keys.include? notice[:error_class]
end
```

If the block returns true the notice is sent; false it's ignored. So this says "ignore the notice if the class is a rescue response".

File: config/honeybadger.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+---
+api_key: <%= ENV['HONEYBADGER_API_KEY'] %>
```
Comment (joshuap) [inline config/honeybadger.yml]: You could delete this file entirely now because HONEYBADGER_API_KEY will be picked up automatically from ENV at runtime.

**Comment by joshuap** [comment]
One other thing to note (since you're configuring the API key via ENV): by default Honeybadger will start in development and test environments, but it uses a null backend which doesn't send any live data. This assumes that an API key will be available in those environments (it's not a big deal if there isn't; it will just log that it can't start). If you want it to start (so that it simulates most of what it does in production locally) you can add a dummy api key to those environments.

**Comment by Ch4s3** [comment]
@joshuap Thanks! Your first comment confirms what I thought was the case based on the docs. I'll try your tweak and pull out the `api_key: <%= ENV['HONEYBADGER_API_KEY'] %>`.

**Comment by arthurnn** [comment]
@Ch4s3 I think this looks good now.. Can you squash your commits in one, as this changed direction a lot.

thanks

**Comment by Ch4s3** [comment]
@arthurnn all set.

**Comment by arthurnn** [comment]
thanks

**Comment by Ch4s3** [comment]
:) always glad to help.

**Pull request merged**